setuptools/command/bdist_wheel.py

```diff
@@ -134,7 +134,13 @@ def get_abi_tag() -> str | None:
 
 
 def safer_name(name: str) -> str:
-    return safe_name(name).replace("-", "_")
+    return (
+        # Per https://packaging.python.org/en/latest/specifications/name-normalization/#name-normalization
+        re.sub(r"[-_.]+", "-", safe_name(name))
+        .lower()
+        # Per https://packaging.python.org/en/latest/specifications/binary-distribution-format/#escaping-and-unicode
+        .replace("-", "_")
+    )
```

Comment:
I'm not a fan of this duplicated code being further expanded. I'd like to see this logic consolidated (in setuptools if not in something like packaging), especially since now it's more than one line that's being duplicated.

Reply:
Resolved in 9f764e8.